### DISCUSSION TITLE
System taking too much load

### DISCUSSION THREAD

**Discussion by Parth-Maniar-3012** [filamentphp/filament#6653] [General]
Hello, 

I have created one small demo in which I have a student and standard basic CRUD.
but in the filament when I tested that with **500** and **11000** records it will take too much time while selecting **All** in the data table filter and also the system will hang out and nothing will be working properly.

Sometimes I got an error that the execution time exceeds 60 seconds.
So @danharrin @ryangjchandler and the team can you please help to solve this issue?

I have attached a video for better understanding.
https://www.awesomescreenshot.com/video/17861358?key=2a3e3a41b722984d49e33b25fcfe5955

Thank you in advance.

**Comment by danharrin** [comment]
Please don't @ people when asking questions, be patient and wait for someone to answer in their own time.

There is nothing we can solve. Loading 500 records into a table at once is going to be very slow, because there is so much data. You can disable the All option of the pagination to ensure it does not get used.

**Reply by Parth-Maniar-3012** [reply]
What if I want the All option in the data table filter?

Is there any alternative solution?

**Reply by danharrin** [reply]
Do you really think 500 table rows on one page is a good experience for users?

**Reply by Parth-Maniar-3012** [reply]
No, from the user's perspective, it is not a good habit.
I will remove that option I think that is the best idea only.

Thank you!

**Comment by RX-77IT** [comment]
Have you tried disabling the views collector of the debugbar?
If not, go in \config\debugbar.php (you can publish it with `php artisan vendor:publish --provider="Barryvdh\Debugbar\ServiceProvider"` ) and turn off the views:
```php
'collectors' => [
        // ...
        'views' => false,  // Views with their data
        // ...
    ],
```

I always have to disable them in filament because there are a lot of them and the debugbar will start to slow down everything.

**Reply by Parth-Maniar-3012** [reply]
I have tried this but I will remove the view tab from the debugger. but views will be rendered in the same numbers as they were rendered before.

**Comment by yob-yob** [comment]
The issue here is actually on the browser side... it's rendering too much HTML on the browser... and think about the Event Listeners for every actions on every single row....

I don't know if IntersectionObserver can fix this, but you can give it a try... would probably conflict with Livewire, so you need to use wire:ignore for that... but yeah it's kinda hard to find a good solution, so if you can find a way to show all the items only if the users has scrolled to that point then also remove the items that the user has passed scrolling to then it would greatly impact the rendering speed of your website on the user's browser it would also decrease their RAM usage...

**Reply by killjin** [reply]
I think what you want is this, the virtual table: https://www.surely.cool/, unfortunately it is vue, our project has real-time customer service chat, it would be great if there is a vue version.  https://github.com/filamentphp/filament/discussions/6924
